integrationtests/testground/main.go

```diff
@@ -403,49 +422,68 @@ func (t *threadWithKeys) Sharable() *SharedInfo {
 
 // WaitForRecords blocks until it receives nRecords, then return them
 func (t *threadWithKeys) WaitForRecords(ctx context.Context, nRecords int) (records []corenet.Record) {
-	ch, err := t.cli.Subscribe(ctx, corenet.WithSubFilter(t.Info.ID))
-	if err != nil {
-		msg("Error subscribing thread: %v", err)
-		return
-	}
-	debug("Subscribed to thread")
-	total := 0
-	for record := range ch {
-		total++
+	msg("Waiting for %d unique records", nRecords)
+	for record := range t.subscribeCh {
+		rec := record.Value()
+		if _, exists := t.seenRecords[rec.Cid()]; exists {
```

Comment:
This was meant to deduplicate records but there's no longer duplication with the fix